src/docfx/template/TemplateEngine.cs

```diff
@@ -43,16 +46,75 @@ private TemplateEngine(string templateDir, JsonSchema metadataSchema)
                 .ToDictionary(prop => prop.Key, prop => prop.Value.HtmlMetaName);
         }
 
-        public static (JsonSchemaValidator, JsonSchemaTransformer) GetJsonSchema(Schema schema)
+        public static SourceInfo<string> ReadMimeFromFile(string pathToDocset, string filePath)
+        {
+            SourceInfo<string> mime = default;
+
+            if (filePath.EndsWith(".json", PathUtility.PathComparison))
+            {
+                if (File.Exists(filePath))
+                {
+                    using (var reader = new StreamReader(filePath))
+                    {
+                        mime = JsonUtility.ReadMime(reader, pathToDocset);
+                    }
+                }
+            }
+            else if (filePath.EndsWith(".yml", PathUtility.PathComparison))
+            {
+                if (File.Exists(filePath))
+                {
+                    using (var reader = new StreamReader(filePath))
+                    {
+                        mime = new SourceInfo<string>(YamlUtility.ReadMime(reader), new SourceInfo(pathToDocset, 1, 1));
+                    }
+                }
+            }
+
+            return mime;
+        }
+
+        public static string GetSchemaName(string mime)
```

Comment:
`schemaName` is the same as `mime`, so this method might be unnecessary. We can consolidate the term used internally to always be `schemaName`, could also remove OutputModel.SchemaType